Niak 61 Login and logout

Sewerinius opened the pull request.

Sewerinius committed:
[NIAK-55] Added spring-security dependency

Sewerinius committed:
[NIAK-55] Better docker gradle caching

Sewerinius committed:
[NIAK-55] Added Spring Security dependency

Sewerinius committed:
[NIAK-55] Added Spring Security configuration

Sewerinius committed:
[NIAK-55] Updated storing and loading users

Sewerinius committed:
[NIAK-55] Secured all enpoints

Sewerinius committed:
[NIAK-55] Disabled the DesignForExtension check

Sewerinius committed:
[NIAK-55] Set Wildcard import check to warning level

Sewerinius committed:
[NIAK-55] Added Spring Security Test and fixed one test

Sewerinius committed:
[NIAK-55] Merge remote-tracking branch 'remotes/origin/develop' into NIAK-55

Sewerinius committed:
[NIAK-55] A little refactoring

Sewerinius committed:
[NIAK-62] Added Redux

Sewerinius committed:
[NIAK-62] Added user reducer

Sewerinius committed:
[NIAK-62] Moved RouteProvider

Sewerinius committed:
[NIAK-62] Added routes reducer

Sewerinius committed:
[NIAK-62] Added reselect

Sewerinius committed:
[NIAK-62] Selecting Routes

Sewerinius committed:
[NIAK-62] User session persistence

Sewerinius committed:
[NIAK-62] Merge remote-tracking branch 'remotes/origin/develop' into NIAK-62

Sewerinius committed:
[NIAK-55] Merge remote-tracking branch 'remotes/origin/develop' into NIAK-55

# Conflicts:
#	rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/controller/RecruitmentPeriodController.java
#	rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/controller/SystemUserController.java
#	rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/service/CourseOfStudyService.java
#	rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/service/UserService.java

Sewerinius committed:
[NIAK-55] Secured newest Controllers

Sewerinius committed:
[NIAK-62] Missing routes fix

Sewerinius committed:
[NIAK-61] Merge branch 'NIAK-55' into NIAK-61

Sewerinius committed:
Fixed permission problems

Sewerinius committed:
[NIAK-61] Login page, login and logout functionality

Sewerinius committed:
[NIAK-61] Checkstyle fixes

Sewerinius committed:
[NIAK-61] Merge remote-tracking branch 'remotes/origin/develop' into NIAK-61

jakubowiczish commented on rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/config/security/DenyByDefaultMethodSecurityMetadataSource.java:
```diff
@@ -0,0 +1,41 @@
+    @Override
+    protected Collection<ConfigAttribute> findAttributes(Method method, Class<?> targetClass) {
+        MergedAnnotations methodAnnotations = MergedAnnotations.from(method);
+        if (targetClass.getPackageName().startsWith("org.niewidoczniakademicy.rezerwacje")
```
I would split these all statemets into separate methods

jakubowiczish commented on rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/controller/SystemUserController.java:
```diff
@@ -49,31 +57,46 @@ public GetSystemUsersResponse getSystemUsersByFirstNameAndLastName(@RequestParam
+    @ResponseStatus(value = HttpStatus.OK)
+    public Object getMe() {
+        Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+        System.out.println(auth);
```
Redundant already, right?

jakubowiczish commented on rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/model/security/UserPrincipal.java:
```diff
@@ -0,0 +1,54 @@
+        return true;
+    }
+
+    public SystemUser getUser() {
```
Can be replaced with Lombok's ```@Getter```

Sewerinius committed:
[NIAK-55] Merge branch 'develop' into NIAK-55

# Conflicts:
#	rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/controller/ConnectionController.java
#	rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/controller/CourseOfStudyController.java

Sewerinius committed:
[NIAK-55] Secured more endpoints

Sewerinius committed:
[NIAK-62] Merge branch 'develop' into NIAK-62

Sewerinius committed:
[NIAK-62] Import fix

Sewerinius committed:
[NIAK-62] Merge remote-tracking branch 'origin/develop' into NIAK-62

Sewerinius committed:
[NIAK-55] Merge remote-tracking branch 'remotes/origin/develop' into NIAK-55

# Conflicts:
#	rezerwacje-app/config/checkstyle/checkstyle.xml
#	rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/controller/ConnectionController.java
#	rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/controller/RecruitmentController.java

Sewerinius committed:
[NIAK-55] Secure controllers

Sewerinius committed:
[NIAK-61] Merge branch 'NIAK-55' into NIAK-61

# Conflicts:
#	rezerwacje-front/src/common/utilities/RouteProvider.jsx

Sewerinius committed:
[NIAK-61] Merge branch 'NIAK-62' into NIAK-61
